### PR TITLE
docs: add React examples for checkbox

### PR DESCRIPTION
**Pull request by tomivirkki**
Extracted from https://github.com/vaadin/docs/pull/2632

### CODE REVIEW

File: frontend/demo/component/checkbox/react/checkbox-custom-presentation.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import React, { useEffect, useState } from 'react';
+import { CheckboxGroup } from '@hilla/react-components/CheckboxGroup.js';
+import { Checkbox } from '@hilla/react-components/Checkbox.js';
+import { getPeople } from 'Frontend/demo/domain/DataService';
+import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
+
+function Example() {
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/react/checkbox-custom-presentation.tsx:8]: The "custom presentation" example seems to be not used anywhere. Maybe we should remove it.

File: frontend/demo/component/checkbox/react/checkbox-edit-multiple.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import React from 'react';
+import { Checkbox } from '@hilla/react-components/Checkbox.js';
+
+function Example() {
+  return <Checkbox label="Enabled" />;
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/react/checkbox-edit-multiple.tsx:6]: The "edit multiple" example seems to be a stub from https://github.com/vaadin/docs/pull/29. Consider removing it.

File: frontend/demo/component/checkbox/react/checkbox-custom-presentation.tsx (hunk 1 above)
Comment (tomivirkki) [reply frontend/demo/component/checkbox/react/checkbox-custom-presentation.tsx]: done

File: frontend/demo/component/checkbox/react/checkbox-edit-multiple.tsx (hunk 2 above)
Comment (tomivirkki) [reply frontend/demo/component/checkbox/react/checkbox-edit-multiple.tsx]: done

File: frontend/demo/component/checkbox/react/checkbox-trigger.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { reactExample } from 'Frontend/demo/react-example'; // hidden-source-line
+import React from 'react';
+import { Checkbox } from '@hilla/react-components/Checkbox';
+
+function Example() {
+  return <Checkbox label="Enabled" />;
```
[the remark this answers is not in the document]
Comment (tomivirkki) [reply frontend/demo/component/checkbox/react/checkbox-trigger.tsx]: done

**Review by web-padawan — APPROVED** [review]

**Pull request merged**